### PR TITLE
[#2] [Android] Update app to follow Android Compose template structure

### LINKED ISSUE #2: [Android] Update app to follow Android Compose template structure
## Why

We should update the KMM project for Android following the Android Compose template.

## Acceptance Criteria

- Update the KMM project for Android following the Android Compose template https://github.com/nimblehq/android-templates.

### PR DESCRIPTION
**Pull request by luongvo**
- Close #2

## What happened 👀

- Update the KMM project for Android following the [Android Compose template](https://github.com/nimblehq/android-templates).
- Stop using `BuildConfig` from Gradle 8.
- Add signingConfigs for debug and `release` builds.

## Insight 📝

- I followed a good approach to managing Dependencies & Versions from this KMM's sample project [PeopleInSpace](https://github.com/joreilly/PeopleInSpace).
- Gradle does not generate BuildConfig by default. We should stop using and follow the best practices to have custom code in each build source folder. https://developer.android.com/build/releases/past-releases/agp-8-0-0-release-notes#default-changes

## Proof Of Work 📹

Apps should work properly, with no changes in UI.

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -7,6 +7,7 @@ kotlin.code.style=official
 #Android
 android.useAndroidX=true
 android.nonTransitiveRClass=true
+android.defaults.buildfeatures.buildconfig=true
```
Comment (luongvo) [inline gradle.properties]: FYI: this is required to have `BuildConfig` in MyApplication https://developer.android.com/build/releases/past-releases/agp-8-0-0-release-notes#default-changes

File: android/proguard-rules.pro [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+#-keepattributes InnerClasses # Needed for `getDeclaredClasses`.
+#-if @kotlinx.serialization.Serializable class
+#com.example.myapplication.HasNamedCompanion, # <-- List serializable classes with named companions.
+#com.example.myapplication.HasNamedCompanion2
+#{
+#    static **$* *;
+#}
+#-keepnames class <1>$$serializer { # -keepnames suffices; class is kept when serializer() is kept.
+#    static <1>$$serializer INSTANCE;
+#}
```
Comment (luongvo) [inline android/proguard-rules.pro:62]: FYI: we will need these rules when integrating the networking layer with Ktor.

File: buildSrc/src/main/java/Configurations.kt [hunk 3]
```diff
@@ -0,0 +1,32 @@
+object Flavors {
```
Comment (phongvhd93) [inline buildSrc/src/main/java/Configurations.kt:1]: @luongvo, I think we have a duplicate works with @kaungkhantsoe's works in this [PR](https://github.com/nimblehq/kmm-templates/pull/25)

Comment (luongvo) [reply buildSrc/src/main/java/Configurations.kt:1]: @phongvhd93 Yes, we can rebase for the following PRs. I think this PR should go first 🤷‍♂️ cc @kaungkhantsoe

**Review by phongvhd93 — APPROVED** [review]

File: android/src/main/java/co/nimblehq/kmm/template/android/ui/navigation/AppNavigation.kt [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    navController: NavHostController = rememberNavController(),
+    startDestination: String = AppDestination.Home.destination
+) {
+    NavHost(
+        navController = navController,
+        startDestination = startDestination
+    ) {
+        composable(AppDestination.Home) {
+            HomeScreen(
+                // TODO
```
Comment (kaungkhantsoe) [inline android/src/main/java/co/nimblehq/kmm/template/android/ui/navigation/AppNavigation.kt:19]: Shall we add some sentence here?

**Review by kaungkhantsoe — APPROVED** [review]
Just one minor optional concern. The rest LGTM 🚀

File: android/src/main/java/co/nimblehq/kmm/template/android/ui/navigation/AppNavigation.kt (hunk 4 above)
Comment (luongvo) [reply android/src/main/java/co/nimblehq/kmm/template/android/ui/navigation/AppNavigation.kt:19]: @kaungkhantsoe Yes, however, let's add more initial examples in the following tasks where we added an initial MVVM flow from data > domain > app level.

**Comment by luongvo** [comment]
@phongvhd93 👋

**Pull request merged**